core/src/main/java/org/apache/iceberg/PartitionSpecParser.java

```diff
@@ -138,6 +140,7 @@ private static void buildFromJsonFields(PartitionSpec.Builder builder, JsonNode
         "Cannot parse partition spec fields, not an array: %s", json);
 
     Iterator<JsonNode> elements = json.elements();
+    int fieldIdCnt = 0;
```

Comment:
Nit: we like to avoid abbreviations that aren't readable. Saving a few chars isn't worth trading readability.